pkg/cmd/auth/status/status.go

```diff
@@ -164,3 +171,26 @@ func statusRun(opts *StatusOptions) error {
 
 	return nil
 }
+
+func userAndScopes(httpClient *http.Client) (string, string, error) {
+	url := fmt.Sprintf("%suser", ghinstance.RESTPrefix(ghinstance.Default()))
```

Comment:
Hi, the `user` endpoint isn't available when using the generated GITHUB_TOKEN in Actions. The API returns an error: `resource not available by integration` and the command aborts.

On the other hand, fetching the viewer via GraphQL works and returns `Logged in to github.com as github-actions[bot]`.

Also note that we already have a mechanism of getting scopes: https://github.com/cli/cli/blob/fa2223129b536efad8704a66c270630b47dc8b1d/pkg/cmd/auth/shared/oauth_scopes.go#L34

Reply:
@mislav thank you! I wasn't thrilled with the implementation I had, this one is much better.

In the earlier line in the command https://github.com/cli/cli/blob/6bf194f63e3cf6de0ea69d99fb0d3c9a726f7c50/pkg/cmd/auth/status/status.go#L100 we are fetching the scopes and discarding them, which necessitates a second API call to get them. What do you think about changing the signature of https://github.com/cli/cli/blob/6bf194f63e3cf6de0ea69d99fb0d3c9a726f7c50/pkg/cmd/auth/shared/oauth_scopes.go#L63 to return `(string, error)` where string is the list of scopes from the api?

It would save an extra api call and make the tests less repetitive. A quick code search shows this function is only called in a few places.

Reply:
One thing I noticed in testing, in https://github.com/cli/cli/blob/6bf194f63e3cf6de0ea69d99fb0d3c9a726f7c50/pkg/cmd/auth/status/status.go#L124 and the one I added below it https://github.com/cli/cli/blob/6bf194f63e3cf6de0ea69d99fb0d3c9a726f7c50/pkg/cmd/auth/status/status.go#L129 we don't bubble up errors to the command, we add them to the output message. 

This meant that in some of the tests I would see unmatched http stubs in the output but the test conditions of not having an error and matching parts of the regular expression message we ok, and so the test passed.

Reply:
@mntlty I like the idea of keeping it to just a single api request for the scope headers. What do you think about creating a new function `HeaderHasMinimumScopes(scopeHeader string) error` that would be called from `HasMinimumScopes` and also the `auth status` command with the extracted scopes header so that we don't have to change the function signature of `HasMinimumScopes`?

As for the tests, we probably should add `defer reg.Verify(t)` to these tests so that they fail on unmatched http stubs.